### PR TITLE
[build]: put stretch debian packages under target/debs/stretch/

### PR DESCRIPTION
**Pull request by lguohan**
debian packages for stretch are separated from jessie debian packages

Signed-off-by: Guohan Lu <gulv@microsoft.com>

**- What I did**

**- How I did it**
- in stretch build phase, all debian packages built in that stage are placed under target/debs/stretch directory.
- for python-based debian packages, since they are really the same for jessie and stretch, they are placed under target/python-debs directory.

**- How to verify it**
build the image and test

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
@lguohan: Could you please provide a brief description of the new intended package directory structure and build process so that I can better review the changes?

**Comment by NStetskovych** [comment]
what about other platforms? despite mlnx and vs?

**Comment by lguohan** [comment]
@NStetskovych, it looks cavium platform did not put the kernel drivers into stretch directory. I fixed the problem, however, I do not really have a way to validate the image. can you check?

**Comment by nathanalderson** [comment]
I believe this PR introduced a new build failure on top of a (long-existing) build failure in the p4 platform. See https://sonic-jenkins.westus2.cloudapp.azure.com/job/p4/job/buildimage-p4-all/950/.

File: Makefile.work [hunk 1]
```diff
@@ -103,6 +103,7 @@ DOCKER_BUILD = docker build --no-cache \
 
 SONIC_BUILD_INSTRUCTION :=  make \
                            -f slave.mk \
+                           BLDENV=$(BLDENV) \
```
Comment (qiluo-msft) [inline Makefile.work]: >BLDENV [](start = 27, length = 6)

slave.mk could detect the BLDENV from /etc/os-release. We should not treat it as a parameter.

File: slave.mk [hunk 2]
```diff
@@ -24,10 +24,18 @@ SRC_PATH = src
 RULES_PATH = rules
 TARGET_PATH = target
 DOCKERS_PATH = dockers
+ifdef BLDENV
```
Comment (qiluo-msft) [inline slave.mk]: >BLDENV [](start = 6, length = 6)

We should assume BLDENV always defined. either STRETCH or JESSIE, and SID in future.

If keep backward compatibility, we could create target/debs as symlink to target/debs/jessie

File: Makefile.work (hunk 1 above)
Comment (lguohan) [reply Makefile.work]: good suggestion, I'll try later.

File: slave.mk (hunk 2 above)
Comment (qiluo-msft) [reply slave.mk]: Actually backward compatibility should not be an issue. It is easy to fix build script or jenkins script. If this is true, we don't need a symlink.

---
In reply to: [275602303](https://github.com/Azure/sonic-buildimage/pull/2519#discussion_r275602303) [](ancestors = 275602303)

**Pull request merged**